src/services/status.service.js

```diff
@@ -1,28 +1,152 @@
+/**
+ * Functions for getting and setting status values
+ * @module services/status
+ */
+
 const {
   getStoredStatus,
-  updateStoredStatus
+  updateStoredStatus,
+  getEmailDistribution
 } = require("../connectors/status/status-db.connector");
 
+const { logEmitter } = require("./logging.service");
+const { NOTIFY_STATUS_TEMPLATE, ENVIRONMENT_DESCRIPTION } = require("../config");
+const { sendSingleEmail } = require("../connectors/notify/notify.connector");
+
+/**
+ * Function that returns the status
+ *
+ * @param {string} statusName The name of status field to return
+ *
+ * @returns {object} All status values for the specified field name
+ */
 const getStatus = async statusName => {
+  logEmitter.emit(
+    "functionCall",
+    "status.service",
+    "getStatus"
+  );
+
   const status = await getStoredStatus();
+  
+  logEmitter.emit(
+    "functionSuccess",
+    "status.service",
+    "getStatus"
+  );
+
   return statusName ? status[statusName] : status;
 };
 
+/**
+ * Updates a specified status value
+ *
+ * @param {string} statusName The status field name
+ * @param {any} newStatus The new status value
+ *
+ * @returns {any} The new status value
+ */
 const setStatus = async (statusName, newStatus) => {
+  logEmitter.emit(
+    "functionCall",
+    "status.service",
+    "setStatus"
+  );
+
+  const status = await getStoredStatus();
+  const currentValue = status[statusName];
   const updatedStatusValue = await updateStoredStatus(statusName, newStatus);
+
+  if(newStatus !== currentValue) {
+    let statusText = "Invalid";
+    if(newStatus === true) {
+      statusText = "Succeeded";
+    } else if(newStatus === false) {
+      statusText = "Failed";
+    }
+
+    /**
+     * Format status name to be more readable for use in email
+     * Replace function steps:
+     * 1. Replace first instance of 'Succeeded' with empty string
+     * 2. Replace first instance of 'mostRecent' with empty string
+     * 3. Replace all instances of capital letters with itself prefixed with a space (e.g. 'newStatusName' => 'new Status Name')
+     */
+    let formattedStatusName = statusName.replace('Succeeded','').replace('mostRecent','').replace(/([A-Z])/g, ' $1').toLowerCase().trim();
+    const data = {
+        environment_description: ENVIRONMENT_DESCRIPTION,
+        status_name: formattedStatusName.charAt(0).toUpperCase() + formattedStatusName.slice(1),
```

Comment:
A function for transforming this string with some JsDoc explanation would be nice